cmake/toolchain/zephyr/generic.cmake

```diff
@@ -1,3 +1,8 @@
 # SPDX-License-Identifier: Apache-2.0
 
+if(NOT(DEFINED(ENV{SDK_VERSION})))
+  message(FATAL_ERROR "SDK_VERSION must be set")
+  return ()
+endif()
+
```

Comment:
In what situations is it possible for SDK_VERSION to be unset?

Reply:
i experienced this several times - most likely due to a broken environment. the effect is that the result was in once case (cmake 3.13.4) a segfaulting cmake, and in the other case a cmake error. figuring out the cause of the segmentation fault wasn't obvious.

Reply:
commit message has been updated with this: "This may happen if ZEPHYR_TOOLCHAIN_VARIANT is set to "zephyr", but
ZEPHYR_SDK_INSTALL_DIR is invalid (or unset)."